pyfar/dsp/dsp.py

```diff
@@ -187,3 +187,112 @@ def spectrogram(signal, dB=True, log_prefix=20, log_reference=1,
     times -= times[0]
 
     return frequencies, times, spectrogram
+
+
+def normalize(signal, norm_type='time', operation='max', channelwise='max',
+              value=None, freq_range=(20, 22e3)):
+    """
+    Normalize signal in the time or frequency domain.
+
+    Parameters
+    ----------
+    signal: Signal
+        Input signal of the signal class
+    norm_type: string
+        'time' - Normalize the time signal to 'value'
+        'abs' - Normalize the magnitude spectrum to 'value'
+        'dB' - Normalize the db magnitude spectrum to 'value'
+        The default is 'time'
+    operation: string
+        'max' - Normalize to the absolute maximum of the signal data
+        'mean' - Normalize to the mean of the signal data
+        'rms' - Normalize to the rms of the signal data
+        The default is 'max'
+    channelwise: string
+        'each' - Normalize each channel separately
+        'max' - Normalize to the max or operation across all channels
+        'min' - Normalize to min or operation across all channels
+        'mean' - Normalize to mean of operation across all channels
+       The default is 'max'
+    value:
+        Normalizes to 'value' which can be a scalar or a vector with
+        a number of elements equal to channels
+        The default is 0 for norm_type='db' and 1 otherwise
```

Comment:
```suggestion
        Normalizes to `value` which can be a scalar or a vector with
        a number of elements equal to channels. The unit of `value`
        is defined by `norm_type`, i.e., it is either dB or linear. The
        default is 0 for norm_type='db' and 1 otherwise
```